dataprocessing/data_utils.py

```diff
@@ -0,0 +1,75 @@
+import cv2
+import os
+import logging
+from pathlib import Path
+logging.basicConfig(level=logging.INFO, format='%(levelname)s: %(message)s')
+logger = logging.getLogger(__name__)
+
+
+def videos2frames_frame_number(inputdir: Path, outputdir: Path, frame_number: int):
+    """
+    convert video to frames based on the requested number of frames.
+    """
+    assert inputdir.is_dir()
+    if not outputdir.is_dir():
+        outputdir.mkdir(parents=True, exist_ok=False)
+
+    files = list(x for x in inputdir.iterdir() if x.is_file() and x.suffix == '.mp4')
+    logger.info(f'list of video files are: {files}')
+    for i in range(len(files)):
+        current_output = outputdir / Path(Path(files[i]).name).stem
+        logger.info(f'ouputput folder for frames is: {current_output}')
+        try:
+            os.mkdir(current_output)
+        except OSError:
+            pass
+        logger.info(f'saving the frames for {files[i]}')
+        vidcap = cv2.VideoCapture(str(files[i]))
+        fps = vidcap.get(cv2.CAP_PROP_FPS)
+        fcnt = vidcap.get(cv2.CAP_PROP_FRAME_COUNT)
+        logger.info(f'framerate:, {fps}')
+        logger.info(f'frame count:, {fcnt}')
+        logger.info(f'video duration:, {fcnt/fps}')
+        count = 0
+        save_count = 0
+        while True:
+            success, image = vidcap.read()
+            if not success:
+                break
+            if count % frame_number == 0:
+                cv2.imwrite(str(current_output / "frame{:d}.jpg".format(count)), image)
+                save_count += 1
+                logger.info(f'frame {count} is saved!')
+            count += 1
+
+
+def frame2video_convertor(pathIn: Path, pathOut: Path, fps: int):
```

Comment:
very minor, use snake_case for variable/parameter names

Reply:
done here 
a07481e